src/modules/lifecycle_management.json

```diff
@@ -0,0 +1,254 @@
+{
+    "openrpc": "1.2.4",
+    "info": {
+        "title": "LifecycleManagement",
+        "description": "A module for managing application lifecycle.",
+        "version": "0.0.0"
+    },
+    "components": {
+        "schemas": {
+            "State": {
+                "type": "string",
+                "description": "The application lifecycle state",
+                "enum": [
+                    "initializing",
+                    "inactive",
+                    "foreground",
+                    "background",
+                    "unloading",
+                    "suspended"
+                ]
+            },
+            "LifecycleParameters": {
+                "title": "LifecycleParameters",
+                "type": "object",
+                "required": [
+                  "appId"
+                ],
+                "properties": {
+                  "appId": {
+                    "description": "The application ID",
+                    "type": "string"
+                  }
+                }
+            },        
+            "LifecycleResult": {
```

Comment:
So this is the object we pass to each provider method?

If so, the name `LifecycleResult` (which contains `LifecycleParameters`) is a bit confusing...